src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateActionUtils.java

```diff
@@ -1,35 +1,56 @@
 package com.commercetools.sync.producttypes.utils;
 
 
+import com.commercetools.sync.commons.exceptions.BuildUpdateActionException;
+import com.commercetools.sync.producttypes.ProductTypeSyncOptions;
 import io.sphere.sdk.commands.UpdateAction;
 import io.sphere.sdk.producttypes.ProductType;
 import io.sphere.sdk.producttypes.ProductTypeDraft;
 import io.sphere.sdk.producttypes.commands.updateactions.ChangeDescription;
 import io.sphere.sdk.producttypes.commands.updateactions.ChangeName;
 
 import javax.annotation.Nonnull;
+import java.util.List;
 import java.util.Optional;
 
 import static com.commercetools.sync.commons.utils.CommonTypeUpdateActionUtils.buildUpdateAction;
+import static com.commercetools.sync.producttypes.utils.ProductTypeUpdateAttributeDefinitionActionUtils.buildAttributeDefinitionsUpdateActions;
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
 
 public final class ProductTypeUpdateActionUtils {
+    private static final String PRODUCT_TYPE_CHANGE_NAME_EMPTY_NAME = "Cannot unset 'name' field of product type with "
+        + "id '%s'.";
+    private static final String PRODUCT_TYPE_CHANGE_DESCRIPTION_EMPTY_DESCRIPTION = "Cannot unset 'description' field "
+        +   "of product type with id '%s'.";
 
     /**
      * Compares the {@code name} values of a {@link ProductType} and a {@link ProductTypeDraft}
      * and returns an {@link Optional} of update action, which would contain the {@code "changeName"}
      * {@link UpdateAction}. If both {@link ProductType} and {@link ProductTypeDraft} have the same
      * {@code name} values, then no update action is needed and empty optional will be returned.
      *
+     * <p>Note: If the name of the new {@link ProductTypeDraft} is null, an empty {@link Optional} is returned with no
+     * update actions and a custom callback function, if set on the supplied {@link ProductTypeSyncOptions}, is called.
+     *
      * @param oldProductType the product type that should be updated.
      * @param newProductType the product type draft which contains the new name.
+     * @param syncOptions    the sync syncOptions with which a custom callback function is called in case the parent
+     *                       is null.
      * @return optional containing update action or empty optional if names are identical.
      */
     @Nonnull
     public static Optional<UpdateAction<ProductType>> buildChangeNameAction(
         @Nonnull final ProductType oldProductType,
-        @Nonnull final ProductTypeDraft newProductType) {
-        return buildUpdateAction(oldProductType.getName(), newProductType.getName(),
-            () -> ChangeName.of(newProductType.getName()));
+        @Nonnull final ProductTypeDraft newProductType,
```

Comment:
Nitpick: I would put here a line break, like there is one in the `buildChangeDescriptionAction`, so we have a consistent style, also to make it easier to read by separating it a bit from the params line.